hazelcast/include/hazelcast/client/Socket.h

```diff
@@ -83,6 +84,14 @@ namespace hazelcast {
             virtual client::Address getAddress() const = 0;
 
             virtual void setBlocking(bool blocking) = 0;
+
+            /**
+             *
+             * This function is used to obtain the locally bound endpoint of the socket.
+             *
+             * @returns An address that represents the local endpoint of the socket.
+             */
+            virtual std::auto_ptr<Address> localSocketAddress() const = 0;
```

Comment:
Is it possible for this method to return null ?
If not, value itself is more appropriate. 
I think naming of this method should be either same with java or aligned with the `getAdress` or `getRemoteEndpoint` here.

Reply:
Yes, it can be null if the connection is not connected.

Naming convention:
At Java ClientConnection.getLocalSocketAddress --> C++ Connection::getLocalSocketAddress
At Java channel.localSocketAddress() --> C++ socket->localSocketAddress

Is this not aligned with Java?